### PR TITLE
Add black and codeql

### PR DESCRIPTION
**Pull request by SamTov**
Added the black tests again to overwrite PR #286

### CODE REVIEW

**Review by SamTov — COMMENTED** [review]
It is as in the original PR

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+     branches:
+       - main
+
+ jobs:
+   doc:
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: Black Check
+         uses: psf/black@stable
```
Comment (PythonFZ) [inline .github/workflows/black.yml:17]: missing blank line at the end of the file

File: .github/workflows/codeql_analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ # ******** NOTE ********
+ # We have attempted to detect the languages in your repository. Please check
+ # the `language` matrix defined below to confirm you have the correct set of
+ # supported CodeQL languages.
+ #
+ name: "CodeQL"
+
+ on:
+   push:
+     branches: [ main, development ]
```
Comment (PythonFZ) [inline .github/workflows/codeql_analysis.yml:16]: I don't know if we really wnat CodeQL running on PRs to dev - I think main is sufficient? It takes for ever and as not brought up anyting for us yet.

File: .github/workflows/codeql_analysis.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+     # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+     #    and modify them (or add more) to build your code if your project
+     #    uses a compiled language
+
+     #- run: |
+     #   make bootstrap
+     #   make release
+
+     - name: Perform CodeQL Analysis
+       uses: github/codeql-action/analyze@v1
```
Comment (PythonFZ) [inline .github/workflows/codeql_analysis.yml:71]: missig line at the end of file

**Review by PythonFZ — APPROVED** [review]

**Pull request merged**